internal/operands/tekton-tasks/reconcile.go

```diff
@@ -307,3 +276,18 @@ func getTektonTasksNamespace(request *common.Request) string {
 	}
 	return request.Instance.Namespace
 }
+
+func filterResources[T any](items []T, trimSuffix bool) []T {
+	var results []T
+	for _, item := range items {
+		object, _ := any(item).(metav1.ObjectMeta)
```

Comment:
This is still an unchecked cast. Please return early with empty list at least if that happens.